src/full/Agda/TypeChecking/DeadCode.hs

```diff
@@ -51,7 +51,16 @@ eliminateDeadCode bs disp sig ms = Bench.billTo [Bench.DeadCode] $ do
   let hasCompilePragma =
         Set.fromList . HMap.keys .
         HMap.filter (not . MapS.null . defCompiledRep) $ defs
-      rootNames = Set.union public hasCompilePragma
+
+      isPrimitive = \case
+        Primitive{}     -> True
+        PrimitiveSort{} -> True
+        _ -> False
+      primitives =
+        Set.fromList . HMap.keys .
+        HMap.filter (isPrimitive . theDef) $ defs
```

Comment:
We could join the filtering steps for `hasCompilePragma` and `primitives`  (one run over `defs` only).
Also, is `HMap.filter` sensible?  Shouldn't we rather convert to a list first and then filter?

Reply:
Perhaps converting to a list first is slightly more efficient.

Reply:
Matches my intuition.  Fixed.